src/cudamatrix/cu-matrix.cc

```diff
@@ -2528,8 +2528,40 @@ void CuMatrixBase<Real>::AddElements(Real alpha,
 }
 
 template<typename Real>
+void CuMatrixBase<Real>::AddElements(Real alpha, const CuArray<Int32Pair> &indexes,
+				     const Real *input) {
+  if (indexes.Dim() == 0) return;
+  KALDI_ASSERT(input != NULL);
+  
+#if HAVE_CUDA == 1
+  if (CuDevice::Instantiate().Enabled()) {
+    CuVector<Real> tmp_vec(indexes.Dim(), kUndefined);
+    CU_SAFE_CALL(cudaMemcpy(tmp_vec.Data(), input, indexes.Dim() * sizeof(Real),
+                            cudaMemcpyHostToDevice));
+
+    Timer tim;
```

Comment:
Please move this line "Timer tim;" to the top of the code block; it should measure everything we do in the function.